### PR TITLE
fix(office): stop agent heartbeat cooldowns from being silently ignored

### PR DESCRIPTION
**Pull request by nova28**
> [!TIP]
> **PR walkthrough:** [Open the visual walkthrough](https://walkthrough.kandev.ai/pr/3512/373062afdd6f.html)

**Today:** Office agents configured with a heartbeat cooldown (e.g. "wait N seconds between runs") fire on every heartbeat tick regardless of the configured value — the gate reads `agent_profiles.last_run_finished_at`, a column no code path ever writes.
**After this:** The gate reads `office_agent_runtime.last_run_finished_at`, the column Office actually maintains, so `cooldown_sec` behaves as documented and paces heartbeat-triggered runs.
**Who hits this:** Anyone running an Office agent with `cooldown_sec > 0` — today every heartbeat fires as if cooldown were 0, regardless of the configured value.
**Scope:** standalone bugfix, two changes landed together because they're the same "what does cooldown mean" question (see Important Changes below).
**Not here:** the dead `agent_profiles.last_run_finished_at` column, `cooldown_sec` defaults, the separate auto-pause-after-N-consecutive-failures mechanism, schema/migrations, and the frontend.

Office's heartbeat scheduler is meant to skip an agent whose `cooldown_sec` hasn't elapsed since its last run, but the gate was reading a column that nothing in the codebase ever populates, so the check always saw a zero-value timestamp and let every heartbeat through.

## Important Changes

- `internal/backendapp/cron.go` (`heartbeatAgentRuntime.AllowFire`): read `office_agent_runtime.last_run_finished_at` instead of the dead `agent_profiles` column; `CooldownSec <= 0` still short-circuits to always-allow, and a lookup error still fails closed.
- `internal/office/service/failure.go` (`HandleAgentFailure`): stamp `last_run_finished_at` on a failed run too, so cooldown paces every terminal outcome (success, stop, or failure) the same way. The separate auto-pause-after-N-consecutive-failures mechanism is untouched — this only affects the per-run cooldown clock.

## Validation

- `go test ./internal/backendapp/... ./internal/office/service/... -count=1` → all pass, including two new regression tests that fail red against `origin/main` and pass green on this branch:
  - `TestHeartbeatAgentRuntime_AllowFire_ReadsOfficeRuntimeCooldown`
  - `TestSchedulerTick_AgentFailedReleasesTaskCheckout` (asserts the failure-path stamp)
- `make typecheck test lint` — `typecheck` and `lint` clean (backend `golangci-lint`: 0 issues; web eslint: 0 warnings; harness/spec/architecture lint: all pass). The full `make test` run fails in ~13 unrelated packages (`internal/worktree`, `internal/task/service`, `internal/agentctl/server/process`, etc.) due to concurrent full-suite runs from sibling git worktrees on this machine; reproduced the same failures against the merge-base (`77e3cfc97`) in a scratch worktree, confirming they're pre-existing/environmental, not caused by this diff. `internal/backendapp` and `internal/office/service` — the only packages this diff touches — pass cleanly on both.
- `make lint-format` → clean (Prettier).
- `pnpm run i18n:ratchet` (from `apps/web`) → clean, no UI copy touched.
- `git merge-base HEAD origin/main` is an ancestor already in this branch's history — no rebase needed.

## Possible Improvements

Low-to-moderate risk: this restores real pacing behavior, so any Office agent already configured with `cooldown_sec > 0` will start actually waiting between heartbeat-triggered runs instead of firing every tick. That's the intended fix, but it's a user-visible timing change worth calling out for anyone who (unknowingly) relied on the no-op.

## Checklist

- [ ] If I do not have repository write access and this is a large architectural change, I discussed the direction in a linked issue before opening this PR.
- [x] This PR contains one logical change; unrelated work is split into separate PRs.
- [x] I have performed a self-review of my code.
- [x] I have manually tested my changes and they work as expected.
- [x] My changes have tests that cover the new functionality and edge cases.
- [ ] If my change touches UI files (`apps/web/`), I have added or updated Playwright e2e tests in `apps/web/e2e/` and verified them with `make test-e2e`.
- [x] I checked whether this affects public docs in `docs/public/**` and updated them or noted why no docs change is needed. (No public docs reference this internal scheduling detail.)

<a href="https://cubic.dev/pr/kdlbs/kandev/pull/3512?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: apps/backend/internal/backendapp/cron_test.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+	dbConn, err := db.OpenSQLite(filepath.Join(t.TempDir(), "cron-cooldown.db"))
+	if err != nil {
+		t.Fatalf("open sqlite: %v", err)
+	}
+	database := sqlx.NewDb(dbConn, "sqlite3")
+	t.Cleanup(func() { _ = database.Close() })
+	if _, cleanup, err := store.Provide(database, database, nil); err != nil {
+		t.Fatalf("agent settings store migrations: %v", err)
+	} else {
+		t.Cleanup(func() { _ = cleanup() })
```
[the remark this answers is not in the document]
Comment (nova28) [reply apps/backend/internal/backendapp/cron_test.go]: Fixed in e27963c5c: cleanup is now registered unconditionally before the fatal check, matching the AGENTS.md convention.

File: docs/specs/office/system-design/runtime-01.md [hunk 2]
```diff
@@ -68,6 +68,7 @@ A follow-up spec can introduce per-adapter classification once we have real usag
...
 - The wakeup row is stamped with `status = failed` and `error_message`. No follow-up wakeup is queued from the failure path.
 - Re-runs only happen via explicit user action: **Resume session** in chat, **Mark fixed** on an inbox entry, or task reassignment to a different agent.
+- A terminal failure still stamps `office_agent_runtime.last_run_finished_at` — `HandleAgentFailure` calls the same `Service.stampRunFinished` helper the completed/stopped paths use. Cooldown is a run-pacing mechanism, not a success tracker: a below-threshold failing agent is paced by `cooldown_sec` on its next heartbeat-driven fire exactly as it would be after a success, so a flapping provider or a misconfigured agent cannot retry back-to-back with zero pacing and burn cost/tokens before hitting auto-pause. This applies uniformly across every task assigned to the agent, not just the one that just failed - `cooldown_sec` gates the agent's next fire for any reason. The consecutive-failure counter and auto-pause threshold (below) are the separate, existing mechanism for stopping a chronically broken agent altogether; cooldown only paces the failures that happen before that threshold is reached.
```
[the remark this answers is not in the document]
Comment (nova28) [reply docs/specs/office/system-design/runtime-01.md]: Confirmed - task assignment and manual resume go through ClaimNextRun/ProcessRunGuard, which don't consult last_run_finished_at (TestCooldown_RecentFinish_GuardAllows). Fixed in e27963c5c: narrowed the sentence to heartbeat-triggered fires and named the scheduler path that isn't gated.

File: apps/backend/internal/backendapp/cron.go [hunk 3]
```diff
@@ -175,11 +178,19 @@ func (r *heartbeatAgentRuntime) AllowFire(
...
-		}
+	if agent.CooldownSec <= 0 {
+		return true, nil
+	}
+	runtime, err := r.office.GetAgentRuntime(ctx, agentID)
+	if err != nil {
+		return false, err
+	}
+	if runtime == nil || runtime.LastRunFinishedAt == nil {
+		return true, nil
```
[the remark this answers is not in the document]
Comment (nova28) [reply apps/backend/internal/backendapp/cron.go:189]: Verified real, but deferred rather than fixed here. `stampRunFinished`'s swallow-and-log pattern for `UpdateRuntimeLastRunFinished` predates this PR (introduced in #2902) - this PR is what makes it consequential, since `AllowFire` now actually reads that column. A write failure would let one agent's cooldown fire early for one cycle (reverting to this PR's pre-fix, no-op-cooldown behavior), not cause data loss or a security issue, and matches every other best-effort write in this file (clearAgentWorking, publishRunProcessedForWorkspace). Making the write transactionally reliable or making AllowFire fail closed is a reliability/behavior decision affecting all three completion paths (success/stopped/failed) plus the read side - bigger than a fixup-round change, and needs the same kind of owner decision that shaped this PR's two changes. Filed as a follow-up: https://github.com/kdlbs/kandev/pull/3512 (task 3d231948-e977-4cb5-a17c-17d6df642be1, kept unstarted pending a decision).

**Comment by carlosflorencio** [comment]
Maintainer fixup in 373062a strengthens the cooldown regression coverage: it checks the exact expiry boundary, null runtime timestamps, and fail-closed runtime lookup errors. It also aligns the failure-path timestamp test and plan wording with the implementation. Focused backend tests pass. Thanks for the contribution.
